libraries/Microsoft.Bot.Configuration/BotConfigurationExtensions.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.Configuration
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+
+    public static class BotConfigurationExtensions
+    {
+        /// <summary>
+        /// Returns a IEnumerable of a specific <see cref="ConnectedService"/> type contained in the bot configuration file.
+        /// </summary>
+        /// <param name="botServices">BotConfiguration's services.</param>
+        /// <param name="serviceType">The type of <see cref="ConnectedService"/> to search.</param>
+        /// <returns>List of bot's <see cref="ConnectedService"/>.</returns>
+        public static IEnumerable<ConnectedService> OfType(this List<ConnectedService> botServices, string serviceType)
+        {
+            // If botServices is null, throws an exception
+            if (botServices == null)
+            {
+                throw new ArgumentNullException();
```

Comment:
Should supply `nameof(botServices)` as parameter to the exception so it's clear what argument the exception is being thrown for.